packages/opentelemetry-plugin-http/src/http.ts

```diff
@@ -417,9 +428,27 @@ export class HttpPlugin extends BasePlugin<Http> {
   }
 
   private _startHttpSpan(name: string, options: SpanOptions) {
-    const span = this._tracer
-      .startSpan(name, options)
-      .setAttribute(AttributeNames.COMPONENT, this.component);
+    /*
+     * If a parent is required but not present, we use a `NoRecordingSpan` to still
+     * propagate context without recording it.
+     */
+    const hasParent = this._tracer.getCurrentSpan() !== undefined;
+    const requireParent =
+      options.kind === SpanKind.CLIENT
+        ? this._config.requireParentforOutgoingSpans
+        : this._config.requireParentforIncomingSpans;
+    let span: Span;
+    if (hasParent === false && requireParent === true) {
```

Comment:
Nit: check from line 435 could be done as 2nd param in `if` statement as it doesn't need to be called always as you are checking `requireParent` anyways
for example
`if (requireParent === true && !this._tracer.getCurrentSpan())`

Reply:
@vmarchaud I was going to merge this but we can wait for this change if you want to make it?

Reply:
Sure i'm gonna fix this before the sig

Reply:
Done, you can merge it @dyladan